src/AutoRest.CSharp/Output/Models/Types/InheritanceChoser.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Input;
+using Azure.ResourceManager.Core;
+
+namespace AutoRest.CSharp.Output.Models.Types
+{
+    internal static class InheritanceChoser
+    {
+        public static IList<System.Type> ReferenceClassCollection = GetReferenceClassCollection();
```

Comment:
nit: Type

Reply:
there is a conflict with CSharp.Output.Models.Type